VideoWeb/VideoWeb/Helpers/AllocationHearingsEventNotifier.cs

```diff
@@ -0,0 +1,52 @@
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using VideoApi.Contract.Requests;
+using VideoWeb.EventHub.Handlers.Core;
+using VideoWeb.EventHub.Models;
+using VideoWeb.Helpers.Interfaces;
+using EventType = VideoWeb.EventHub.Enums.EventType;
+
+namespace VideoWeb.Helpers
+{
+    public class AllocationHearingsEventNotifier: IAllocationHearingsEventNotifier
+    {     
+        private readonly IEventHandlerFactory _eventHandlerFactory;
+        private readonly ILogger<AllocationHearingsEventNotifier> _logger;
+
+        public AllocationHearingsEventNotifier(IEventHandlerFactory eventHandlerFactory, ILogger<AllocationHearingsEventNotifier> logger)
+        {
+            _eventHandlerFactory = eventHandlerFactory;
+            _logger = logger;
+        }
+        
+        public Task PushAllocationHearingsEvent(string csoUserName, IList<HearingDetailRequest> hearings)
+        {
+            CallbackEvent callbackEvent = new CallbackEvent()
+            {
+                EventType = EventType.AllocationHearings,
+                TimeStampUtc = DateTime.UtcNow,
+                AllocatedHearingsDetails = hearings.ToList(),
+                CsoAllocatedUserName = csoUserName
+            };
+
+            _logger.LogTrace("Publishing event to UI: {event}", JsonSerializer.Serialize(callbackEvent));
+            return PublishEventToUi(callbackEvent);
+        }
+
+        private Task PublishEventToUi(CallbackEvent callbackEvent)
+        {
+            if (callbackEvent == null)
```

Comment:
If the notifier is called with a null object, then something has gone wrong and we should allow the code to fail so we can capture the error

Reply:
Please see other implementations. I am just following what has been done in other notifier. As always if you think it ihas been done wrong in the other we need a refactor ticket to change all of them

Reply:
Only one has been done this way and one has been done the way suggested. Capturing nulls when there shouldn't be any is making a problem which we don't want otherwise something will go wrong and we won't know

Reply:
Only one has been done this way because only one it is passing the callbackEvent. All the others are passing different objects and initialising the callbackEvent in the notifier.